src/components/scopedVariables/utils.tsx

```diff
@@ -0,0 +1,128 @@
+import yaml from 'js-yaml'
+import { ScopedVariablesDataType } from './types'
+import { FileReaderStatus, ValidatorType } from '../common/hooks/types'
+import {
+    EMPTY_FILE_STATUS,
+    FILE_NOT_SUPPORTED_STATUS,
+    PARSE_ERROR_STATUS,
+    JSON_PARSE_ERROR_STATUS,
+    YAML_PARSE_ERROR_STATUS,
+} from './constants'
+
+export const validator: ValidatorType = ({ data, type }) => {
```

Comment:
Moving it to common may cause issues since it was meant to be for this case only and if someone changed the functionality it would cause breaking issues in scoped variables